# Week 5

### Import Pandas and Data Sources

In [33]:
import pandas as pd

In [34]:
airports = pd.read_csv('https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/airports.csv')
weather = pd.read_csv('https://raw.githubusercontent.com/hadley/nycflights13/master/data-raw/weather.csv')

## Question #1
### Sort airports by lat (latitude) descending and select the top 5 which will be the most norther airports

In [35]:
airports.sort_values('lat', ascending=False).head()

faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt  tz dst              tzone  
417  149  -5   A                NaN  
230   44  -9   A  America/Anchorage  
110   41  -9   A  America/Anchorage  
708   35  -9   A  America/Anchorage  
152   96  -9   A  America/Anchorage

### The results show the northernmost airport is Dillant Hopkins Airport in New Hampshire. 
This doesn't make sense at all as New Hampshire is not even close to being the highest state in the main part of the U.S., forget about Alaska. 
This is clearly a mistake as if you google the latitude and longitude 72.270833	42.898333 its somewhere in the Barents Sea over West Russia visible in this link: 
https://www.google.com/maps/place/72%C2%B016'15.0%22N+42%C2%B053'54.0%22E/@67.2564548,27.7807161,3z/data=!4m13!1m7!3m6!1s0x0:0x0!2zNzLCsDE2JzE1LjAiTiA0MsKwNTMnNTQuMCJF!3b1!8m2!3d72.270833!4d42.898333!3m4!1s0x0:0x0!8m2!3d72.270833!4d42.898333

### The actual northern most airport is Wiley Post-Will Rogers Memorial Airport, 1749 Akavak St, Utqiagvik, AK 99723.
This airport is actually the northernmost airport. You can see the difference between the two airports on this link;
https://www.google.com/maps/dir/Wiley+Post-Will+Rogers+Memorial+Airport,+Akavak+St,+Utqiagvik,+AK/Keene+Dillant-Hopkins+Airport,+80+Airport+Rd,+Swanzey,+NH+03446/@57.5550077,-131.5220156,4z/data=!4m14!4m13!1m5!1m1!1s0x50c327fff620aae7:0x936de8c8fd2d094b!2m2!1d-156.7802701!2d71.2874212!1m5!1m1!1s0x89e173a4cdd956e9:0xed387407375c239!2m2!1d-72.2726019!2d42.9064535!3e0


## Question #2
### Filter airports and select airports where tzone is not null and contains America. Then Sort airports by lon (longitude) descending and selecting the top 5 will return the easetern airports

In [36]:
airports[(airports.tzone.notnull()) & (airports['tzone'].str.contains("America"))].sort_values('lon', ascending=False).head()

faa                                 name        lat         lon  alt  \
1290  SYA                         Eareckson As  52.712275  174.113620   98   
444   EPM           Eastport Municipal Airport  44.910111  -67.012694   45   
624   HUL                         Houlton Intl  46.123083  -67.792056  489   
259   CAR                         Caribou Muni  46.871500  -68.017917  626   
1101  PQI  Northern Maine Rgnl At Presque Isle  46.688958  -68.044797  534   

      tz dst              tzone  
1290  -9   A  America/Anchorage  
444   -5   A   America/New_York  
624   -5   A   America/New_York  
259   -5   A   America/New_York  
1101  -5   A   America/New_York

## This also has a mistake. It shows Eareckson Air Station on the iland of Shemya in the Alaskan Aleutian Islands. 

https://www.google.com/search?q=eareckson+as+airport&rlz=1C1CHBF_enUS827US827&oq=Eareckson+As&aqs=chrome.1.0l5j69i60l2j69i61.1947j0j7&sourceid=chrome&ie=UTF-8
This is obviously a mistake. 

## The correct answer is EPM Eastport Municipal Airport in Maine.

## Question #3
## Merge the two dataframes together

In [37]:
mergedData = weather.merge(airports, left_on='origin', right_on='faa')

Filter and select weather for 2/12/2013 in New York region and sort by Wind Speed

In [38]:
mergedData[(mergedData.year == 2013) & (mergedData.month == 2) & (mergedData.day == 12) & (mergedData.tzone.str.contains("New_York"))].sort_values('wind_speed', ascending=False).head()

origin  year  month  day  hour   temp   dewp  humid  wind_dir  \
1009     EWR  2013      2   12     3  39.02  26.96  61.63     260.0   
18417    LGA  2013      2   12     2  42.98  26.06  50.94     290.0   
1018     EWR  2013      2   12    12  44.06  26.06  48.87     270.0   
18428    LGA  2013      2   12    13  44.06  23.00  43.02     300.0   
18429    LGA  2013      2   12    14  44.06  23.00  43.02     300.0   

       wind_speed  ...  visib             time_hour  faa                 name  \
1009   1048.36058  ...   10.0  2013-02-12T08:00:00Z  EWR  Newark Liberty Intl   
18417    23.01560  ...   10.0  2013-02-12T07:00:00Z  LGA           La Guardia   
1018     21.86482  ...   10.0  2013-02-12T17:00:00Z  EWR  Newark Liberty Intl   
18428    21.86482  ...   10.0  2013-02-12T18:00:00Z  LGA           La Guardia   
18429    20.71404  ...   10.0  2013-02-12T19:00:00Z  LGA           La Guardia   

             lat        lon alt  tz  dst             tzone  
1009   40.692500 -74.168667  18  -5    A  America/New_York  
18417  40.777245 -73.872608  22  -5    A  America/New_York  
1018   40.692500 -74.168667  18  -5    A  America/New_York  
18428  40.777245 -73.872608  22  -5    A  America/New_York  
18429  40.777245 -73.872608  22  -5    A  America/New_York  

[5 rows x 23 columns]

This shows that EWR was the windiest at 1048.36058 wind speed. This is obviously impossible so LGA is the windiest